receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go

```diff
@@ -233,3 +243,28 @@ func (s *scraper) scrapeAndAppendDiskIOMetric(now pcommon.Timestamp, handle proc
 
 	return nil
 }
+
+func (s *scraper) scrapeAndAppendContextSwitchMetrics(now pcommon.Timestamp, handle processHandle) error {
+	switches, err := handle.NumCtxSwitches()
+
+	if err != nil {
+		return err
+	}
+
+	s.mb.RecordProcessVoluntaryCtxSwitchesDataPoint(now, int64(switches.Voluntary))
+	s.mb.RecordProcessInvoluntaryCtxSwitchesDataPoint(now, int64(switches.Involuntary))
+
+	return nil
+}
+
+func (s *scraper) scrapeAndAppendOpenFileDescriptorsMetric(now pcommon.Timestamp, handle processHandle) error {
+	fds, err := handle.NumFDs()
```

Comment:
I'd recommend setting similar conditions as done in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/12886 to avoid throwing errors when metrics are disabled

Reply:
Good call, I was wondering about that. For consistency, I've added conditions for all process metrics, and added a test to check this.